client/lib/wpcom-undocumented/lib/undocumented.js

```diff
@@ -2350,6 +2350,34 @@ Undocumented.prototype.transferStatus = function( siteId, transferId ) {
 };
 
 /**
+ * Update the poster for a video.
+ *
+ * @param {string} videoId ID of the video
+ * @param {object} data The POST data
+ * @param {Function} fn Function to invoke when request is complete
+ * @returns {Promise} A promise that resolves when the request is complete
+ */
+Undocumented.prototype.updateVideoPoster = function( videoId, data, fn ) {
+	debug( '/videos/:video_id/poster' );
+
+	const params = {
+		path: `/videos/${ videoId }/poster`,
+	};
+
+	if ( 'file' in data ) {
+		params.formData = [
+			[ 'poster', data.file ]
+		];
+	}
+
+	if ( 'at_time' in data ) {
```

Comment:
@dbtlr the way we would normally prefer to handle this isn't with a dangling `else` at the end but rather with an early abort from the function call. you will find that we have very few `else` structures in Calypso.

also, in general, I would strongly encourage working together on this to use the new data layer instead of creating a new `wpcom.undocumented` function